File: backend/app/schemas/news.py
from datetime import date

from pydantic import BaseModel, Field


class NewsArticleItem(BaseModel):
    id: int
    source: str
    source_article_id: str
    title: str
    subtitle: str | None = None
    summary: str | None = None
    content: str | None = None
    ministry: str | None = None
    published_date: str
    url: str
    thumbnail_url: str | None = None
    collected_at: str
    updated_at: str | None = None


class NewsSyncRequest(BaseModel):
    target_date: date


class NewsSyncResult(BaseModel):
    target_date: str
    inserted: int = 0
    updated: int = 0
    skipped: int = 0
    failed: int = 0
    total: int = 0
    message: str = "completed"


class NewsJobItem(BaseModel):
    id: int
    job_type: str
    status: str
    detail: str | None = None
    created_at: str
    updated_at: str | None = None


class NewsListResponse(BaseModel):
    articles: list[NewsArticleItem]


class NewsSyncResponse(BaseModel):
    result: NewsSyncResult
    articles: list[NewsArticleItem] = Field(default_factory=list)

File: frontend/src/types/news.ts
export type NewsArticle = {
  id: number;
  source: string;
  source_article_id: string;
  title: string;
  subtitle: string | null;
  summary: string | null;
  content: string | null;
  ministry: string | null;
  published_date: string;
  url: string;
  thumbnail_url: string | null;
  collected_at: string;
  updated_at: string | null;
};

export type NewsListResponse = {
  articles: NewsArticle[];
};

export type NewsSyncResult = {
  target_date: string;
  inserted: number;
  updated: number;
  skipped: number;
  failed: number;
  total: number;
  message: string;
};

export type NewsSyncResponse = {
  result: NewsSyncResult;
  articles: NewsArticle[];
};

export type NewsJob = {
  id: number;
  job_type: string;
  status: string;
  detail: string | null;
  created_at: string;
  updated_at: string | null;
};
